java-client/src/main/java/org/opensearch/client/transport/rest_client/RestClientTransport.java

```diff
@@ -273,16 +271,23 @@ private <ResponseT, ErrorT> ResponseT getHighLevelResponse(
                         // TODO: have the endpoint provide the exception constructor
                         throw new OpenSearchException((ErrorResponse) error);
                     }
-                } catch(MissingRequiredPropertyException errorEx) {
+                } catch (MissingRequiredPropertyException errorEx) {
                     // Could not decode exception, try the response type
                     try {
                         ResponseT response = decodeResponse(statusCode, entity, clientResp, endpoint);
                         return response;
-                    } catch(Exception respEx) {
+                    } catch (Exception respEx) {
                         // No better luck: throw the original error decoding exception
-                        throw new TransportException("Failed to decode error response", new ResponseException(clientResp));
+                        throw new TransportException("Failed to decode error response",
+                                new ResponseException(clientResp));
                     }
                 }
+            } else if (endpoint instanceof Endpoint) {
```

Comment:
The endpoint is always instance of `Endpoint`, this is how it is declared `Endpoint<?, ResponseT, ErrorT> endpoint`, so why do we need that?